components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2118,7 +2119,26 @@ public ServiceProvider getApplication(int applicationId) throws IdentityApplicat
             if (serviceProvider == null) {
                 return null;
             }
-            int tenantID = IdentityTenantUtil.getTenantId(serviceProvider.getOwner().getTenantDomain());
+            return getApplication(applicationId, serviceProvider.getOwner().getTenantDomain());
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementException("Failed to get service provider with id: " + applicationId,
+                    e);
+        } finally {
+            IdentityApplicationManagementUtil.closeConnection(connection);
+        }
+    }
+
+    @Override
+    public ServiceProvider getApplication(int applicationId, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            // Load basic application data
```

Comment:
Missing full stop at the end

Reply:
The work of this PR for Introducing a new method and deprecating the previous method was revert back by another PR https://github.com/wso2/carbon-identity-framework/pull/4157/files#diff-0bd1a832f4495aed005c591463b854143fff7695f721b85b5d52f6fecdfb2f63L2136